lib/ecto/adapters/postgres.ex

```diff
@@ -166,6 +166,19 @@ defmodule Ecto.Adapters.Postgres do
     end
   end
 
+  @impl Ecto.Adapter.Storage
+  def storage_status(opts) do
+    database = Keyword.fetch!(opts, :database) || raise ":database is nil in repository configuration"
+    opts = Keyword.put(opts, :database, @default_maintenance_database)
```

Comment:
should we handle the `:maintenance_database` configuration option like we do in [`storage_up/1`](https://github.com/elixir-ecto/ecto_sql/blob/ee02e79166a238c4d73b2de734f762432daf929d/lib/ecto/adapters/postgres.ex#L130:L131)?

Reply:
Ah, yes, good call. We should read the maintain_database option before.

Reply:
Fixed [here](https://github.com/elixir-ecto/ecto_sql/compare/ee02e79166a238c4d73b2de734f762432daf929d..aab4f7a628c00542cc9bff036a11ae56b1a86abb)